solarwinds_apm/propagator.py

```diff
@@ -108,16 +115,54 @@ def inject(
 
         # Remove any xtrace_options_response stored for ResponsePropagator
         trace_state = W3CTransformer.remove_response_from_sw(trace_state)
-
         setter.set(
             carrier, self._TRACESTATE_HEADER_NAME, trace_state.to_header()
         )
 
+        # Remove any baggage stored for custom transaction naming
+        if baggage_header:
+            baggage_header = self.remove_custom_naming_baggage_header(
+                baggage_header,
+            )
+            setter.set(carrier, self._BAGGAGE_HEADER_NAME, baggage_header)
+
+    def remove_custom_naming_baggage_header(
+        self,
+        baggage_header: str,
+    ) -> str:
+        """Removes values used for custom naming from baggage header created
+        by upstream W3CBaggagePropagator propagator, if present"""
+        baggage_entries: list[str] = split(_DELIMITER_PATTERN, baggage_header)
+        baggage_kvs = {}
+        for entry in baggage_entries:
+            try:
+                e_name, e_value = entry.split("=", 1)
```

Comment:
This is also the same as how W3CBaggagePropagator does it: https://github.com/open-telemetry/opentelemetry-python/blob/b9f31e91a8263bc8effc9abce889d438927d47ec/opentelemetry-api/src/opentelemetry/baggage/propagation/__init__.py#L85